src/typechecker/FStar.TypeChecker.Tc.fst

```diff
@@ -863,15 +863,29 @@ let tc_decl env se: list sigelt * list sigelt * Env.env =
      BU.print1 "Processing %s\n" (U.lids_of_sigelt se |> List.map string_of_lid |> String.concat ", ");
    if Env.debug env Options.Low then
      BU.print1 ">>>>>>>>>>>>>>tc_decl %s\n" (Print.sigelt_to_string se);
-   if se.sigmeta.sigmeta_admit
-   then begin
-     let old = Options.admit_smt_queries () in
-     Options.set_admit_smt_queries true;
-     let result = tc_decl' env se in
-     Options.set_admit_smt_queries old;
-     result
-   end
-   else tc_decl' env se
+   let tc_ses, new_ses, env'
+     = if se.sigmeta.sigmeta_admit
+       then begin
+         let old = Options.admit_smt_queries () in
+         Options.set_admit_smt_queries true;
+         let result = tc_decl' env se in
+         Options.set_admit_smt_queries old;
+         result
+       end
+       else tc_decl' env se in
+   // Typecheck attributes
+   List.map (fun se ->
+      // [Substitute] (defined in Pervasives), is added as attribute
+      // by TcInductive when a type has no projector, and this
+      // happens for some types (see
+      // TcInductive.early_prims_inductives) that are defined before
+      // Substitute even exists.
+      // Thus the hack below, that typecheck attribtues only when
+      // they're not literally [Substitute]. TODO: fix this.
+      let is_substitute attr = attr_substitute = attr in
+      {se with sigattrs =      filter                is_substitute        se.sigattrs
+          @ tc_attributes env (filter (fun attr -> not (is_substitute attr)) se.sigattrs) }
+   ) tc_ses, new_ses, env'
```

Comment:
The attribute `Substitute` is inserted as an attribute on some definitions of `Prims`, but `Substitute` is defined later in `FStar.Pervasives`, thus I have this special case that is not looking very nice...